### PR TITLE
Fix PPS timing detectors record and tag names for PCL

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

Fix to the the problem reported in https://github.com/dmwm/T0/pull/4732#issuecomment-1205772587
So far the HPTDC and the SAMPIC used the same record name `PPSTimingCalibrationRcd` so the PCL picked up the correct tag for the HPTDC, but nothing for the SAMPIC, this PR separates the two. Also there is some renaming done in the tags names.

#### PR validation:

Given that https://github.com/cms-sw/cmssw/pull/38893 is not merged yet, there is not much to test with. Will test here.

In the backport PR I did this:
```
python3 Configuration/DataProcessing/test/RunRepack.py --lfn /eos/cms/store/t0streamer/Data/ALCAPPS/000/355/559/run355559_ls0012_streamALCAPPS_StorageManager.dat 
python3 Configuration/DataProcessing/test/RunExpressProcessing.py --scenario AlCaPPS_Run3 --global-tag 124X_dataRun3_Express_Candidate_2022_08_05_17_14_24 --lfn file:write_PrimDS1_RAW.root --alcareco PPSCalMaxTracks
python3 Configuration/DataProcessing/test/RunAlcaSkimming.py --scenario AlCaPPS_Run3 --global-tag 124X_dataRun3_Express_Candidate_2022_08_05_17_14_24 --lfn=file:output.root --skims PromptCalibProdPPSDiamondSampic
python3 Configuration/DataProcessing/test/RunAlcaHarvesting.py --scenario AlCaPPS_Run3 --global-tag 124X_dataRun3_Express_Candidate_2022_08_05_17_14_24 --lfn=file:output.root --alcapromptdataset PromptCalibProdPPSDiamondSampic --dataset /A/B/C
```

where `124X_dataRun3_Express_Candidate_2022_08_05_17_14_24` includes a new DMD tag
`DropBoxMetadata_v8.1_express`
which has the diff wrt to the prev one: 
https://cern.ch/go/n6ZT

Leads to a config which says
```
process.ALCAHARVESTPPSDiamondSampicTimingCalibration_dbOutput = cms.PSet(
    record = cms.string('PPSTimingCalibrationRcd'),
    tag = cms.string('PPSDiamondSampicCalibration_pcl'),
    timetype = cms.untracked.string('runnumber')
)

process.ALCAHARVESTPPSDiamondSampicTimingCalibration_metadata = cms.PSet(
    record = cms.untracked.string('PPSTimingCalibrationRcd_SAMPIC')
)

process.ALCAHARVESTPPSTimingCalibration_dbOutput = cms.PSet(
    record = cms.string('PPSTimingCalibrationRcd'),
    tag = cms.string('PPSDiamondTimingCalibration_pcl'),
    timetype = cms.untracked.string('runnumber')
)

process.ALCAHARVESTPPSTimingCalibration_metadata = cms.PSet(
    record = cms.untracked.string('PPSTimingCalibrationRcd_HPTDC')
)
```
which makes me confident that it will work on the T0 side too.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Needs to be backported to 124X

### CODE REVIEW

**Comment by tvami** [comment]
type ctpps,bugfix

**Comment by tvami** [comment]
urgent 
 * backport needed for data taking

**Comment by tvami** [comment]
test parameters:
  - workflows = 1041, 1043, 1044
  - pull_request  = https://github.com/cms-sw/cmssw/pull/38893

**Comment by tvami** [comment]
@cmsbuild , please test

File: Configuration/StandardSequences/python/AlCaHarvesting_cff.py [hunk 1]
```diff
@@ -248,15 +248,15 @@
 
 # --------------------------------------------------------------------------------------
 # PPS calibration
-ALCAHARVESTPPSTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd'))
+ALCAHARVESTPPSTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd_HPTDC'))
```
Comment (mmusich) [inline Configuration/StandardSequences/python/AlCaHarvesting_cff.py:251]: This is now mismatched with 
https://github.com/cms-sw/cmssw/blob/eb2d3a59a3aa2a722da88f97059810aa841d35c1/CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLHarvester.cc#L150

File: Configuration/StandardSequences/python/AlCaHarvesting_cff.py [hunk 2]
```diff
@@ -248,15 +248,15 @@
...
 ALCAHARVESTPPSTimingCalibration_dbOutput = cms.PSet(record = cms.string('PPSTimingCalibrationRcd'),
                                                     tag = cms.string('PPSDiamondTimingCalibration_pcl'),
                                                     timetype = cms.untracked.string('runnumber')
                                                     )
 
-ALCAHARVESTPPSDiamondSampicTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd'))
+ALCAHARVESTPPSDiamondSampicTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd_SAMPIC'))
```
Comment (mmusich) [inline Configuration/StandardSequences/python/AlCaHarvesting_cff.py:257]: same as above

**Comment by tvami** [comment]
Thanks @mmusich !

**Comment by tvami** [comment]
@cmsbuild , please test

File: CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
     toPut = cms.VPSet(
         cms.PSet(
             record = cms.string('PPSTimingCalibrationRcd'),
-            tag = cms.string('DiamondSampicCalibration'),
```
Comment (mmusich) [inline CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py:30]: I think you might want to change the record name here too.

Comment (tvami) [reply CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py:30]: I have a few problems with that file, like the 
https://github.com/cms-sw/cmssw/blob/fc6a963a563c3f1079b918c81868a24dca9b90b3/CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py#L22

or
https://github.com/cms-sw/cmssw/blob/fc6a963a563c3f1079b918c81868a24dca9b90b3/CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py#L48

or
https://github.com/cms-sw/cmssw/blob/fc6a963a563c3f1079b918c81868a24dca9b90b3/CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py#L38

so I think I'll just suggest PPS to delete that file as it is (in a follow-up PR, which is less urgent)

Comment (mmusich) [reply CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py:30]: Mmh, ok, but why changing it here?

Comment (tvami) [reply CalibPPS/TimingCalibration/test/DiamondSampicCalibrationHarvester_cfg.py:30]: I fixed the record name

**Comment by tvami** [comment]
@cmsbuild , please test

File: Configuration/StandardSequences/python/AlCaHarvesting_cff.py [hunk 4]
```diff
@@ -248,15 +248,15 @@
 
 # --------------------------------------------------------------------------------------
 # PPS calibration
-ALCAHARVESTPPSTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd'))
+ALCAHARVESTPPSTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd_HPTDC'))
 ALCAHARVESTPPSTimingCalibration_dbOutput = cms.PSet(record = cms.string('PPSTimingCalibrationRcd'),
```
Comment (malbouis) [inline Configuration/StandardSequences/python/AlCaHarvesting_cff.py:252]: wouldn't you have to change the record to `PPSTimingCalibrationRcd_HPTDC` here as well?

File: Configuration/StandardSequences/python/AlCaHarvesting_cff.py [hunk 5]
```diff
@@ -248,15 +248,15 @@
...
                                                     tag = cms.string('PPSDiamondTimingCalibration_pcl'),
                                                     timetype = cms.untracked.string('runnumber')
                                                     )
 
-ALCAHARVESTPPSDiamondSampicTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd'))
+ALCAHARVESTPPSDiamondSampicTimingCalibration_metadata = cms.PSet(record = cms.untracked.string('PPSTimingCalibrationRcd_SAMPIC'))
 ALCAHARVESTPPSDiamondSampicTimingCalibration_dbOutput = cms.PSet(record = cms.string('PPSTimingCalibrationRcd'),
```
Comment (malbouis) [inline Configuration/StandardSequences/python/AlCaHarvesting_cff.py:258]: and here to `PPSTimingCalibrationRcd_SAMPIC`?

**Comment by tvami** [comment]
Now that the relvals PR is merged, I managed to run 1043.0 locally, it works now.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by malbouis** [comment]
+alca

- tests pass
- PR in line with description: renames HPTDC and SAMPIC PPS Diamond detectors records to fix metadata and therefore tier0 replay failure.

**Comment by tvami** [comment]
+db

**Comment by tvami** [comment]
@cms-sw/orp-l2 we are essentially fully signed

**Comment by qliphy** [comment]
+1

**Pull request merged**